src/spu/spu.cc

```diff
@@ -907,8 +882,12 @@ void PCSX::SPU::impl::SetupThread() {
     bEndThread = 0;  // init thread vars
     bThreadEnded = 0;
     bSpuInit = 1;  // flag: we are inited
-
-    hMainThread = std::thread([this]() { MainThread(); });
+    // Threads are expensive and causes Thread init and switch overheads
+    // hMainThread = std::thread([this]() { MainThread(); });
+    main_t_rst = std::async([this]() {
```

Comment:
I just realized that this `std::future` is never being `get` anywhere, which probably means the code will never run...

Reply:
I made the` future` a global variable to preserve its value otherwise if the value miss, the Emu hangs. because the program waits till the end of the endless while loop.